cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -176,6 +184,27 @@ protected void startUp() throws Exception {
                                           AppMetadataStore.create(context).getApplicationCount());
     metricsCollectionService.getContext(Collections.emptyMap()).gauge(Constants.Metrics.Program.APPLICATION_COUNT,
                                                                       count);
+
+    Map<ImmutablePair<String, String>, Long> pluginCounts = TransactionRunners.run(transactionRunner,
+                                                            (TxCallable<Map<ImmutablePair<String, String>, Long>>)
+                                                              context ->
+                                                              getPluginCounts(AppMetadataStore.create(context)));
+    for (Map.Entry<ImmutablePair<String, String>, Long> entry : pluginCounts.entrySet()) {
+      Map<String, String> tags = ImmutableMap.of(Constants.Metrics.Tag.PLUGIN_NAME, entry.getKey().getFirst(),
```

Comment:
Is this the name that the user gives the plugin? Or the name of the artifact? 

It should be the second one hopefully